internal/release/main.go

```diff
@@ -0,0 +1,218 @@
+// package main gets the next semver version for the given package using git history
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	git "github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/storer"
+	"golang.org/x/mod/modfile"
+)
+
+func getProjectPath() (string, error) {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		return "", fmt.Errorf("cannot get current file path")
+	}
+	return filepath.Join(filepath.Dir(filename), "..", ".."), nil
+}
```

Comment:
nit: more of a personal preference/style thing

also, no reason to return a wrapped error from `os.Getwd()` imo because the existing returned error is pretty clear



```suggestion
func getProjectPath() (string, error) {
	wd, err := os.Getwd()
	if err != nil {
		return "", err
	}
	return filepath.Dir(filepath.Dir(wd)), nil
}
```

Reply:
Ah yeah I figured this would let people run that from any directory, but I guess we don't do that in go anyway